src/utils.ts

```diff
@@ -128,3 +130,153 @@ export function aggregationToPresentation(aggregation: Aggregation): Presentatio
   }
   return result;
 }
+
+export function replaceVariables(
+  requestSpec: Partial<RequestSpec> | undefined,
+  scopedVars?: ScopedVars
+): Partial<RequestSpec> {
+  const result: typeof requestSpec = {};
+  const t = getTemplateSrv();
+  if (requestSpec === undefined) {
+    return result;
+  }
+  (Object.keys(requestSpec) as Array<keyof Partial<RequestSpec>>).forEach(function <
```

Comment:
I'd like a for-loop more here.

Reply:
why? please sell it to me :-) i don't see a problem here.
i think i implemented it for better typing support, and i fear that this typing support is going to vanish when changed to a for loop.

Reply:
This is probably only about what I'm used to, but when I see a `forEach`, is something that is done for side effects alone, while here we're producing another value in `result`. So I'll change my opinion now and say that a `map` over `Object.entries(requestSpec)` would be more appropriate here, but I guess that would make typing even harder.

I'm not sure what type information you would lose with a for loop here. Your loop variable would have type `keyof Partial<RequestSpec>` and the rest should be the same.

Anyway, I don't see this as a blocker. If rewriting would be too much effort, I'd be happy to think about it the next time I stumble across it.